core/domain/voiceover_domain.py

```diff
@@ -0,0 +1,219 @@
+            raise utils.ValidationError(
+                'entity_id must be a string, recieved %r' % self.entity_id)
+        if not isinstance(self.entity_version, int):
+            raise utils.ValidationError(
+                'entity_version must be an int, recieved %r' %
+                self.entity_version)
+        if not isinstance(self.language_accent_code, str):
+            raise utils.ValidationError(
+                'language_accent_code must be a string, recieved %r' %
+                self.language_accent_code)
+
+        for content_id, voiceover_type_to_voiceover in self.voiceovers.items():
+            if not isinstance(content_id, str):
+                raise utils.ValidationError(
+                    'content_id must be a string, recieved %r' % content_id)
+            for voiceover_type, voiceover in (
+                    voiceover_type_to_voiceover.items()):
+                if not isinstance(voiceover_type, str):
+                    raise utils.ValidationError(
+                        'voiceover type must be string, recieved %r' %
+                        voiceover_type)
+                if voiceover_type not in ['manual', 'auto']:
+                    raise utils.ValidationError(
+                        'The voiceover type must be either manual or auto, but '
+                        'received %r' % voiceover_type)
+                voiceover.validate()
+
+    def add_voiceover(
+        self,
+        content_id: str,
+        voiceover_type: str,
+        voiceover: state_domain.Voiceover
+    ) -> None:
+        """Adds voiceover to the entity voiceover instance."""
+        if not isinstance(content_id, str):
+            raise utils.ValidationError(
+                'content_id must be a string, received %r' % content_id)
+
+        if voiceover_type not in ['manual', 'auto']:
+            raise utils.ValidationError(
+                'The voiceover type must be either manual or auto, but '
+                'received %r' % voiceover_type)
+        voiceover.validate()
+        self.voiceovers[content_id][voiceover_type] = voiceover
+
+    def remove_voiceover(
+        self,
+        content_id: str,
+        voiceover_type: str
+    ) -> None:
+        """Removes voiceover from the entity voiceover instance."""
+        if not isinstance(content_id, str):
+            raise utils.ValidationError(
+                'content_id must be a string, received %r' % content_id)
+
+        if voiceover_type not in ['manual', 'auto']:
+            raise utils.ValidationError(
+                'The voiceover type must be either manual or auto, but '
+                'received %r' % voiceover_type)
+
+        del self.voiceovers[content_id][voiceover_type]
+
+    @classmethod
+    def create_empty(
+        cls,
+        entity_id: str,
+        entity_type: str,
+        entity_version: int,
+        language_accent_code: str
+    ) -> EntityVoiceover:
+        """Creates a new and empty EntityVoiceover object."""
```

Comment:
Creates a new, empty EntityVoiceover object.

(reason: "and" mildly suggests two things).

Reply:
Done, thank you for explaining.